subsys/bluetooth/host/smp.c

```diff
@@ -962,7 +962,7 @@ static void sc_derive_link_key(struct bt_smp *smp)
 
 static void smp_br_reset(struct bt_smp_br *smp)
 {
-	k_delayed_work_cancel(&smp->work);
+	k_work_cancel_delayable(&smp->work);
```

Comment:
I also noticed the lack of explicit handling of the return values, but figured that since it wasn't done before, there wouldn't be any real need to change it here. Saw that one of the calls had that added, so perhaps it would make sense to add a (void) cast of the return value to the others as well.